### PR TITLE
Adding background-color and border-width variables to HR element

### PR DESCRIPTION
**Pull request by devhoussam**
**Fixes** [#33793](https://github.com/twbs/bootstrap/issues/33793)

### CODE REVIEW

**Comment by ffoodd** [comment]
Thanks for the PR :)

I don't see the point of this: `currentColor` makes inheriting color useful (so if one override this variable, `$hr-color` doesn't make sense) and resetting `border` is, well, a reset.

I don't see any use case for this, how would it be useful?

**Comment by devhoussam** [comment]
> Thanks for the PR :)
> 
> I don't see the point of this: `currentColor` makes inheriting color useful (so if one override this variable, `$hr-color` doesn't make sense) and resetting `border` is, well, a reset.
> 
> I don't see any use case for this, how would it be useful?

I think it is important because it exists and is already used and it would be better to edit it by using variables only.

**Comment by ffoodd** [comment]
Why editing it at all? What's the use case?

File: scss/_reboot.scss [hunk 1]
```diff
@@ -65,8 +65,8 @@ body {
 hr {
   margin: $hr-margin-y 0;
   color: $hr-color; // 1
-  background-color: currentColor;
-  border: 0;
+  background-color: $hr-background-color;
```
Comment (mdo) [inline scss/_reboot.scss:68]:
```suggestion
  background-color: $hr-bg-color;
```

File: scss/_variables.scss [hunk 2]
```diff
@@ -502,6 +502,8 @@ $blockquote-footer-font-size: $small-font-size !default;
 
 $hr-margin-y:                 $spacer !default;
 $hr-color:                    inherit !default;
+$hr-background-color:         currentColor !default;
```
Comment (mdo) [inline scss/_variables.scss:505]:
```suggestion
$hr-bg-color:                 currentColor !default;
```

**Review by mdo — APPROVED** [review]

**Comment by mdo** [comment]
Pushed to https://github.com/twbs/bootstrap/pull/35123.

**Pull request closed without merging**